pallets/democracy-precompiles/src/lib.rs

```diff
@@ -0,0 +1,277 @@
+			// 	return Self::is_selected_candidate(&input[SELECTOR_SIZE_BYTES..]);
+			// }
+			// [0xc9, 0xf5, 0x93, 0xb2] => {
+			// 	//TODO Do we need to verify that there were no additional bytes passed in here?
+			// 	return Self::min_nomination();
+			// }
+			// [0x97, 0x99, 0xb4, 0xe7] => {
+			// 	return Self::points(&input[SELECTOR_SIZE_BYTES..]);
+			// }
+
+			// If not an accessor, check for dispatchables. These calls ready for dispatch below.
+			[0x56, 0xfd, 0xf5, 0x47] => Self::propose(&input[SELECTOR_SIZE_BYTES..])?,
+			// [0xb7, 0x69, 0x42, 0x19] => Self::leave_candidates()?,
+			// [0x76, 0x7e, 0x04, 0x50] => Self::go_offline()?,
+			// [0xd2, 0xf7, 0x3c, 0xeb] => Self::go_online()?,
+			// [0x28, 0x9b, 0x6b, 0xa7] => Self::candidate_bond_less(&input[SELECTOR_SIZE_BYTES..])?,
+			// [0xc5, 0x7b, 0xd3, 0xa8] => Self::candidate_bond_more(&input[SELECTOR_SIZE_BYTES..])?,
+			// [0x82, 0xf2, 0xc8, 0xdf] => Self::nominate(&input[SELECTOR_SIZE_BYTES..])?,
+			// [0xe8, 0xd6, 0x8a, 0x37] => Self::leave_nominators()?,
+			// [0x4b, 0x65, 0xc3, 0x4b] => Self::revoke_nomination(&input[SELECTOR_SIZE_BYTES..])?,
+			// [0xf6, 0xa5, 0x25, 0x69] => Self::nominator_bond_less(&input[SELECTOR_SIZE_BYTES..])?,
+			// [0x97, 0x1d, 0x44, 0xc8] => Self::nominator_bond_more(&input[SELECTOR_SIZE_BYTES..])?,
+			_ => {
+				log::trace!(
+					target: "democracy-precompile",
+					"Failed to match function selector in democracy precompile"
+				);
+				//TODO test for this
+				return Err(ExitError::Other(
+					"No democracy wrapper method at given selector".into(),
+				));
+			}
+		};
+
+		let outer_call: Runtime::Call = inner_call.into();
+		let info = outer_call.get_dispatch_info();
+
+		// Make sure enough gas
+		if let Some(gas_limit) = target_gas {
+			let required_gas = Runtime::GasWeightMapping::weight_to_gas(info.weight);
+			if required_gas > gas_limit {
+				return Err(ExitError::OutOfGas);
+			}
+		}
+		log::trace!(target: "democracy-precompile", "Made it past gas check");
+
+		// Dispatch that call
+		let origin = Runtime::AddressMapping::into_account_id(context.caller);
+
+		log::trace!(target: "democracy-precompile", "Gonna call with origin {:?}", origin);
+
+		match outer_call.dispatch(Some(origin).into()) {
+			Ok(post_info) => {
+				let gas_used = Runtime::GasWeightMapping::weight_to_gas(
+					post_info.actual_weight.unwrap_or(info.weight),
+				);
+				Ok(PrecompileOutput {
+					exit_status: ExitSucceed::Stopped,
+					cost: gas_used,
+					output: Default::default(),
+					logs: Default::default(),
+				})
+			}
+			Err(e) => {
+				log::trace!(
+					target: "democracy-precompile",
+					"Democracy call via evm failed {:?}",
+					e
+				);
+				Err(ExitError::Other(
+					"Democracy call via EVM failed".into(),
+				))
+			}
+		}
+	}
+}
+
+/// Parses an H160 account address from a 256 bit (32 byte) buffer. Only the last 20 bytes are used.
+fn parse_account(input: &[u8]) -> Result<H160, ExitError> {
```

Comment:
Do we not have something like this elsewhere? This feels like it could easily be a common util fn somewhere (and potentially take a stronger type, e.g. a U256)

Reply:
Thanks to @nanocryk this is all cleaned and unified now :)